packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js

```diff
@@ -91,34 +72,21 @@ describe('Marketplace page - layout', () => {
     const trackUsage = jest.fn();
     useTracking.mockImplementationOnce(() => ({ trackUsage }));
 
-    const { container } = setup();
+    const { queryByText, getByRole } = render();
     await waitForReload();
-    // Check snapshot
-    expect(container.firstChild).toMatchSnapshot();
     // Calls the tracking event
     expect(trackUsage).toHaveBeenCalledWith('didGoToMarketplace');
     expect(trackUsage).toHaveBeenCalledTimes(1);
-    const offlineText = screen.queryByText('You are offline');
-    expect(offlineText).toEqual(null);
+
+    expect(queryByText('You are offline')).toEqual(null);
     // Shows the sort button
-    const sortButton = screen.getByRole('combobox', { name: /Sort by/i });
-    expect(sortButton).toBeVisible();
+    expect(getByRole('combobox', { name: /Sort by/i })).toBeVisible();
     // Shows the filters button
-    const filtersButton = screen.getByText(/Filters/i).closest('button');
-    expect(filtersButton).toBeVisible();
-  });
-
-  it('renders the offline layout', async () => {
-    useNavigatorOnLine.mockReturnValueOnce(false);
-    const { getByText } = setup();
-
-    const offlineText = getByText('You are offline');
-
-    expect(offlineText).toBeVisible();
+    expect(getByRole('button', { name: 'Filters' })).toBeVisible();
```

Comment:
FYI we used to do getByRole here (and in lots of places in this test suite), but we removed it because it had a huge performance cost. Did you not have this issue here?

Reply:
I didn't see anything like that, if it had performance costs I think we should have written a bug on the library because that's very weird. But arguably we're trying to E2E test in unit tests so it's a bit of a tough stand point 🤷🏼‍♀️

Reply:
I remember these were a few issues:

https://github.com/testing-library/dom-testing-library/issues/820
https://github.com/testing-library/dom-testing-library/issues/552#issuecomment-625172052
https://github.com/testing-library/dom-testing-library/issues/698

but yes it's kind of expected since byRole does a lot more checks than other selectors, and as you said we're doing way too much in these tests.

performance aside byRole is super convenient and good for accessibility though. So if it works out without here this is great! but last year at some point they took more than a minute to run, so we had to do something about it (like #14935 and #15006)

Reply:
In the latest run they are quite slow: https://github.com/strapi/strapi/actions/runs/6169977297/job/16745635278?pr=18053#step:7:514

What do you think about creating a follow-up to convert these tests to end-to-end tests in playwright, which is what they are? :)

Reply:
I don't think they're any slower than normal – see this run from what's just been merged into `main` – https://github.com/strapi/strapi/actions/runs/6170121713/job/16746078379#step:7:510